castai/resource_security_runtime_rule.go

```
@@ -0,0 +1,432 @@
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceSecurityRuntimeRuleImporter,
+		},
+
+		Description: "Manages a CAST AI security runtime rule.",
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(3 * time.Minute),
+			Read:   schema.DefaultTimeout(3 * time.Minute),
+			Delete: schema.DefaultTimeout(3 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			FieldRuntimeRuleName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Unique name of the runtime security rule. Name is used as resource identifier in Terraform.",
+				ForceNew:    true, // update is not supported
+			},
+			FieldRuntimeRuleType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Type of the rule (internal value).",
+			},
+			FieldRuntimeRuleCategory: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Category of the rule.",
+				ForceNew:    true, // update is not supported
+			},
+			FieldRuntimeRuleSeverity: {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Severity of the rule. One of SEVERITY_CRITICAL, SEVERITY_HIGH, SEVERITY_MEDIUM, SEVERITY_LOW, SEVERITY_NONE.",
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(supportedSeverities, true)),
+			},
+			FieldRuntimeRuleEnabled: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Whether the rule is enabled.",
+				Default:     false,
+			},
+			FieldRuntimeRuleRuleText: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "CEL rule expression text.",
+			},
+			FieldRuntimeRuleRuleEngineType: {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "The engine type used to evaluate the rule. One of RULE_ENGINE_TYPE_CEL, RULE_ENGINE_TYPE_GO.",
```

For engine we can set default value to RULE_ENGINE_TYPE_CEL and for now hide RULE_ENGINE_TYPE_GO option

---

Fixed.